ipyleaflet/external/here.py

```diff
@@ -90,24 +93,30 @@ def build_tiles_url(**kwargs):
         "For more information please type 'help(build_tiles_url)'.")
 
 
-def basemap_to_tiles(basemap, **kwargs):
+def basemap_to_tiles(basemap, config={}, **kwargs):
     """Return a TileLayer instance with information for HERE map tiles.
 
+    :param basemap: A dictionary with key/value pairs describing valid combinations
+                    of attributes for HERE Location Services map tiles.
+    :param config:  A dictionary with key/value pairs to configure the basemap.
+    :param kwargs:  Keyword arguments passed to TileLayer instance when creating
+                    it.
+    :return: A TileLayer instance.
+
     Example:
 
-    >>> basemap_to_tiles(apikey="foobar")
-    {'url': 'https://3.base.maps.ls.hereapi.com/maptile/2.1/maptile/newest/normal.day/{z}/{x}/{y}/256/png8?lg=eng&apiKey=foobar',
-     'min_zoom': 1,
-     'max_zoom': 18,
-     'attribution': '&copy; <a href="https://here.com">HERE.com</a>',
-     'name': 'HERE'}
+    >>> basemap_to_tiles(basemaps.Default, config={"apikey": "foobar"}).url
+    'https://4.base.maps.ls.hereapi.com/maptile/2.1/maptile/newest/normal.day/{z}/{x}/{y}/256/png8?lg=eng&apiKey=foobar'
     """
+    basemap2 = {}
+    basemap2.update(basemap)
+    basemap2.update(config)
     return TileLayer(
-        url=build_tiles_url(**basemap, **kwargs),
```

Comment:
I would have a personal preference for:
```python
url=build_tiles_url(**basemap, **config, **kwargs)
```
Instead of using a local variable.

Although feel free to ignore this comment if you like it the way you did it.

Reply:
Ok. for the `**config`, but the `**kwargs` are meant for `TileLayer(...)` only or am I missing something?

Reply:
Yes :)

Reply:
Ok now?

Reply:
Not ready yet.

Reply:
Changed to `url=build_tiles_url(**{**basemap, **config})`.